OMCompiler/Compiler/runtime/settingsimpl.c

```diff
@@ -86,14 +93,20 @@ const char* SettingsImpl__getInstallationDirectoryPath(void) {
 const char* SettingsImpl__getInstallationDirectoryPath(void) {
   struct stat sb;
   static char omhome[PATH_MAX];
+  const char *omhome_env = getenv("OPENMODELICAHOME");
   static int init = 0;
   ssize_t r;
   /* This is bad code using hard-coded limits; but we cannot query the size of symlinks on /proc
    * because that FS is not POSIX-compliant.
    */
+  if (omhome_env != NULL)
+    strcpy(omhome, omhome_env);
```

Comment:
This should not be used on Linux. If you have OPENMODELICAHOME set to /usr but call /home/marsj/OpenModelica/build/bin/omc, you will now use the incorrect libraries...

Reply:
> This should not be used on Linux. If you have OPENMODELICAHOME set to /usr but call /home/marsj/OpenModelica/build/bin/omc, you will now use the incorrect libraries...

@sjoelund And let's say a program from /home/adrpo/mine/myProgram loads libOpenModelicaCompiler, what should the getInstallationPath return?

I think that setting OPENMODELICAHOME is an explicit choice of the user, and we should use it.

Reply:
It's unused on Linux and does not do anything. If anything, add a new call to explicitly set the OPENMODELICAHOME for that other application (which might be needed since you can't easily query the location of the so-file when linked to some other executable).
You need some way of knowing this is not the regular executable.

For Linux (specifically), you could use the GNU extension `dladdr` which works fine for SettingsImpl__getInstallationDirectoryPath to query the so-file it was loaded from.
For OSX, I guess you need something else?

Reply:
Of course is not unused anymore on Linux or Mac or Windows as I just used it with my additions :)
If nobody sets it, which is the default, the usual behavior happens.

Reply:
We could also try to fetch it from the exe first and if that doesn't work, use OPENMODELICAHOME if set. But that would not be homogeneous on all platforms and users would not have a way to set what installation directory they want.

Reply:
It seems `dladdr` is used on OSX already. So maybe we could switch the Linux implementation to use that (with _GNU_SOURCE set). Then OPENMODELICAHOME should not be needed on Linux/OSX even for foreign exe linking to OMC.

Reply:
Sounds good, but we need a solution that is sort of homogeneous on all platforms if possible. I will add a setInstallationDirectory API function so one can set it. Another idea would be to add it as an argument to omc_Main_init or omc_Main_main.

Reply:
There seemed to be a function to set it, but all it does is set the env.var. It should also set a flag that this will override the directory on (all? those without auto-detection?) platforms.

Reply:
I guess we can do this automatic deduction of OPENMODELICAHOME from the libOpenModelicaCompiler.(dll|so), even in Windows is possible to get the path. 
I will change the Linux, Mac and Windows versions to try to get the path to libOpenModelicaCompiler. If that doesn't work for some reason, then check OPENMODELICAHOME.

So it will be something like this:
1. by default get the path from the dll|so and use that as OPENMODELICAHOME
2. users can force it via --omhome="path/to/omc" when calling omc_Main_init